### PR TITLE
bpo-35502: Fix reference leaks in ElementTree.TreeBuilder.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue35502

### CODE REVIEW

File: Lib/test/test_xml_etree_c.py [hunk 1]
```diff
@@ -133,6 +134,22 @@ def test_setstate_leaks(self):
         self.assertEqual(len(elem), 1)
         self.assertEqual(elem[0].tag, 'child')
 
+    def test_iterparse_leaks(self):
+        # Test reference leaks in TreeBuilder (issue #35502).
```
Comment (vstinner) [inline Lib/test/test_xml_etree_c.py:138]: Maybe explain that the test is written to be executed using python3 -m test -R to ensure that the code doesn't leak?

File: Misc/NEWS.d/next/Library/2018-12-14-23-56-48.bpo-35502.gLHuFS.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Fixed reference leaks in :class:`xml.etree.ElementTree.TreeBuilder` in case
+of unfinished building of the tree.
```
Comment (vstinner) [inline Misc/NEWS.d/next/Library/2018-12-14-23-56-48.bpo-35502.gLHuFS.rst]: I'm not sure that the error is restricted to "unfinished building of the tree", it's more generally caused by syntax errors: https://bugs.python.org/issue35502#msg332051

**Comment by serhiy-storchaka** [comment]
Thank you @vstinner for your review. I tried to address your comments.

**Review by vstinner — APPROVED** [review]
LGTM.

**Pull request merged**